src/Gzero/Cms/Http/Controllers/Api/BlockFileController.php

```diff
@@ -0,0 +1,162 @@
+<?php namespace Gzero\Cms\Http\Controllers\Api;
+
+use Gzero\Cms\Repositories\BlockReadRepository;
+use Gzero\Core\Http\Controllers\ApiController;
+use Gzero\Core\Http\Resources\FileCollection;
+use Gzero\Core\Jobs\SyncFiles;
+use Gzero\Core\Models\File;
+use Gzero\Core\Validators\FileValidator;
+use Illuminate\Http\Request;
+
+/**
+ * Class BlockFileController
+ *
+ * @SWG\Tag(
+ *   name="blocks",
+ *   description="Everything about app blocks"
+ *   )
+ */
+class BlockFileController extends ApiController
+{
+    /** @var BlockReadRepository */
+    protected $repository;
+
+    /** @var FileValidator */
+    protected $validator;
+
+    /** @var Request */
+    protected $request;
+
+    /**
+     * BlockController constructor.
+     *
+     * @param BlockReadRepository $repository Block repository
+     * @param FileValidator       $validator  Content validator
+     * @param Request             $request    Request object
+     */
+    public function __construct(BlockReadRepository $repository, FileValidator $validator, Request $request)
+    {
+        $this->validator  = $validator->setData($request->all());
+        $this->repository = $repository;
+        $this->request    = $request;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @SWG\Get(
+     *   path="blocks/{id}/files",
+     *   tags={"blocks"},
+     *   summary="List of files synced with block",
+     *   description="List of files synced with block",
+     *   produces={"application/json"},
+     *   security={{"AdminAccess": {}}},
+     *   @SWG\Parameter(
+     *     name="id",
+     *     in="path",
+     *     description="Id of block for which files need to be returned.",
+     *     required=true,
+     *     type="integer"
+     *   ),
+     *   @SWG\Response(
+     *     response=200,
+     *     description="Successful operation",
+     *     @SWG\Schema(type="array", @SWG\Items(ref="#/definitions/File")),
+     *  ),
+     *   @SWG\Response(
+     *     response=422,
+     *     description="Validation Error",
+     *     @SWG\Schema(ref="#/definitions/ValidationErrors")
+     *  ),
+     *   @SWG\Response(response=404, description="Block or files not found"),
+     * )
+     *
+     * @param int $id Block id.
+     *
+     * @return FileCollection
+     *
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function index(int $id)
+    {
+        $block = $this->repository->getById($id);
+        if (!$block) {
+            return $this->errorNotFound();
+        }
+        $this->authorize('read', $block);
+
+        $this->authorize('readList', File::class);
+        $files = $block->files->load('translations');
```

Comment:
We should use BlockReadRepository to get files for specified block instead of using relation on the model